### ISSUE TITLE
dotnet build fails when project 1 is referencing 2 and 3 AND 2 is referencing 3

### ISSUE THREAD

**Issue by christianbernasko** [dotnet/cli#9999]
## Steps to reproduce
tl;dr: when project 1 is referencing 2 and 3 AND 2 is referencing 3 also, `dotnet build` fails

Use case is:
- repository for module `a` is at the root folder
- the `csproj` for a is in src/a/
- `a` is using the `testlib`-library and is therefore referencing it in the libraries folder
- `a` is also using the `b`-library and is referencing it too from there
- since `testlib` is such a useful thing, it is also being referenced by `b` but with its own local copy

The reason for doing it this way is that `b` and `testlib` in the `a`-module are Git submodules. `testlib` in `b` is also a Git submodule in `b`.

Download [repro.zip](https://github.com/dotnet/cli/files/2381192/repro.zip) and try running `dotnet build` for the solution or the `a`-project or the `repro`-solution.

## Expected  behavior
It should compile.

## Actual behavior
Fails with:
```C:\Program Files\dotnet\sdk\2.1.401\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(129,5): error NETSDK1007: Cannot find project info for 'T:\a\libraries\b-repo\libraries\testlib-repo\src\testlib\testlib.csproj'. This can indicate a missing project reference. [T:\a\src\a\a.csproj]```

## Environment data
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.401
 Commit:    91b1c13032

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.401\

Host (useful for support):
  Version: 2.1.3
  Commit:  124038c13e

.NET Core SDKs installed:
  2.1.300 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.401 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

**Comment by livarcocc** [comment]
Not sure what to do here.

In the assets.json file, I see that the libraries session has a reference to the transitive version of the testlib.csproj file. Meanwhile, in the projectReferences one has a reference to the actual project reference.

They both have the same name and in the targets session, we refer to both as testlib/1.0.0, so... I don't think we can disambiguate in this case and we can't really assume that two different projects, in two different locations are the same because they have the same name.

@nguerrera @dsplaisted any ideas if there is anything we could do here?

**Comment by christianbernasko** [comment]
@livarcocc I hope I understood you correctly as I do not know what the assets.json file is. I am not sure if I might made a mistake in creating the repro. Actually, `testlib` the very same Git repository (and therefore the very same project) in both `libraries`-folders. I hope this was clear. For this repro case, I just executed `dotnet new classlib -n testlib` twice which might explain the different session versions.

**Comment by livarcocc** [comment]
@christianbernasko right. They have the same name, so we end up considering them the same but one project references them from one path and the other from another. This causes the problems you are seeing. In this case, if you reference the same project from both places, it should work.

**Comment by christianbernasko** [comment]
@livarcocc Got it, okay. Sure, it would work if it would be in the same place. I hope my use case is clear: it's basically independent libraries which out of coincidence refer to another library and `git-submodule` it.

**Comment by livarcocc** [comment]
@christianbernasko I understand your scenario, but I would say this is not really supported. As far as the build system is concerned, you have the same dependency (dll) coming from two different places and we don't really know how to reconcile that.

**Comment by christianbernasko** [comment]
@livarcocc That I do understand. I am totally aware of that the library should actually be a NuGet package which would resolve the situation.

Thanks for your answer!
